packages/augur-tools/src/flash/fork.ts

```diff
@@ -0,0 +1,74 @@
+import { NULL_ADDRESS } from '../constants';
+import { BigNumber } from 'bignumber.js';
+import { ContractAPI } from '..';
+import { MarketInfo } from '@augurproject/sdk/build/state/getter/Markets';
+import { MarketTypeName } from '@augurproject/sdk/build/state/logs/types';
+
+export async function fork(user: ContractAPI, market: MarketInfo): Promise<boolean> {
+  const MAX_DISPUTES = 20;
+  const SOME_REP = new BigNumber(1e18).times(6e7);
+
+  const payoutNumerators = getPayoutNumerators(market, new BigNumber(0));
+  const conflictNumerators = getPayoutNumerators(market, new BigNumber(1));
+
+  await user.repFaucet(SOME_REP);
+
+  // Get past the market time, into when we can accept the initial report.
+  await user.setTimestamp(new BigNumber(market.endTime + 1));
+
+  const marketContract = user.augur.contracts.marketFromAddress(market.id);
+
+  // Do the initial report, creating the first dispute window.
+  await user.doInitialReport(marketContract, payoutNumerators, '', SOME_REP.toString());
+
+  // Contribution (dispute) fulfills the first dispute bond,
+  // pushing into next dispute round that takes additional stake into account.
+  await user.contribute(marketContract, conflictNumerators, SOME_REP);
+
+  for (let i = 0; i < MAX_DISPUTES; i++) {
+    if ((await marketContract.getForkingMarket_()) !== NULL_ADDRESS) {
+      return true; // forked successfully!
+    }
+
+    const disputeWindow = user.augur.contracts.disputeWindowFromAddress(await marketContract.getDisputeWindow_());
+
+    // Enter the dispute window.
+    await user.setTimestamp((await disputeWindow.getStartTime_()).plus(1));
+
+    // Contribute aka dispute. Opposing sides to keep raising the stakes.
+    const numerators = i % 2 === 0 ? conflictNumerators : payoutNumerators;
+    await user.contribute(marketContract, numerators, SOME_REP);
+  }
+
+  return false; // failed to fork
+}
+
+function getPayoutNumerators(market: MarketInfo, outcome: BigNumber): BigNumber[] {
```

Comment:
why not call util to get the payout numerators?

Reply:
I got an error that looked like a cyclic import. I can make the sdk use these utils to avoid duplication.

Reply:
there is already a util that converts outcome to payout numerators in sdk. use that one. I have a PR up that fixes converting scalar outcome to payout numerators